internal/filter/filter.go

```diff
@@ -107,6 +107,59 @@ func (f *Filter) FindModulesForNode(node client.Object) []reconcile.Request {
 	return reqs
 }
 
+func (f *Filter) FindManagedClusterModulesForCluster(cluster client.Object) []reconcile.Request {
+	logger := f.logger.WithValues("managedcluster", cluster.GetName())
+
+	reqs := make([]reconcile.Request, 0)
+
+	logger.Info("Listing all ManagedClusterModules")
+
+	mods := kmmv1beta1.ManagedClusterModuleList{}
+
+	if err := f.client.List(context.Background(), &mods); err != nil {
+		logger.Error(err, "could not list ManagedClusterModules")
+		return reqs
+	}
+
+	logger.Info("Listed ManagedClusterModules", "count", len(mods.Items))
+
+	clusterLabelsSet := labels.Set(cluster.GetLabels())
+
+	for _, mod := range mods.Items {
+		logger := logger.WithValues("ManagedClusterModule name", mod.Name)
+
+		logger.V(1).Info("Processing ManagedClusterModule")
+
+		sel := labels.NewSelector()
+
+		for k, v := range mod.Spec.Selector {
+			logger.V(1).Info("Processing selector item", "key", k, "value", v)
+
+			requirement, err := labels.NewRequirement(k, selection.Equals, []string{v})
+			if err != nil {
+				logger.Error(err, "could not generate requirement: %v", err)
+				return reqs
+			}
+
+			sel = sel.Add(*requirement)
+		}
+
+		if !sel.Matches(clusterLabelsSet) {
+			logger.V(1).Info("Cluster labels do not match the ManagedClusterModule's selector; skipping")
+			continue
+		}
+
+		nsn := types.NamespacedName{Name: mod.Name, Namespace: mod.Namespace}
```

Comment:
```suggestion
		nsn := types.NamespacedName{Name: mod.Name}
```

Reply:
Oh nice catch! I applied this change :+1: